src/servers/grpc_server.cc

```diff
@@ -1559,8 +1559,8 @@ SharedMemoryControlHandler::Process(Handler::State* state, bool rpc_ok)
     if (request.has_register_()) {
       err = smb_manager_->Create(
           &smb, request.register_().name(),
-          request.register_().shared_memory_key(), request.register_().offset(),
-          request.register_().byte_size());
+          request.register_().system_shared_memory().shared_memory_key(),
```

Comment:
For now, should add a check on 'system_shared_memory' is set, return unimplemented error if 'cuda_shared_memory' is set instead.

Reply:
Thanks! Added a has_system_shared_memory check